gateway/gateway_test.go

```diff
@@ -414,11 +414,15 @@ var _ = Describe("Gateway", func() {
 				err = gateway.StartWebHandler(ctx)
 				Expect(err).To(BeNil())
 			}()
-			getEndpoing, postEndpoints, deleteEndpoints := getEndpointMethodMap()
-
-			verifyEndpoing(getEndpoing, http.MethodGet)
-			verifyEndpoing(postEndpoints, http.MethodPost)
-			verifyEndpoing(deleteEndpoints, http.MethodDelete)
+			require.Eventually(GinkgoT(), func() bool {
+				verifyEndpoint([]string{"/version"}, http.MethodGet)
+				return true
+			}, time.Second*10, time.Second)
+
+			getEndpoint, postEndpoints, deleteEndpoints := getEndpointMethodMap()
+			verifyEndpoint(getEndpoint, http.MethodGet)
+			verifyEndpoint(postEndpoints, http.MethodPost)
+			verifyEndpoint(deleteEndpoints, http.MethodDelete)
 			cancel()
```

Comment:
Here we cancel the context, but we don't wait for `StartWebHandler` to return, before completing the test.